source/braille.py

```diff
@@ -839,25 +839,18 @@ def _isMultiline(self):
 		# Terminals are inherently multiline, so they don't have the multiline state.
 		return (self.obj.role == controlTypes.ROLE_TERMINAL or controlTypes.STATE_MULTILINE in self.obj.states)
 
-	def _getCursor(self):
-		"""Retrieve the collapsed cursor.
-		This should be the start or end of the selection returned by L{_getSelection}.
-		@return: The cursor.
-		"""
-		try:
-			return self.obj.makeTextInfo(textInfos.POSITION_CARET)
-		except:
-			return self.obj.makeTextInfo(textInfos.POSITION_FIRST)
-
 	def _getSelection(self):
 		"""Retrieve the selection.
-		The start or end of this should be the cursor returned by L{_getCursor}.
+		If there is no selection, retrieve the collapsed cursor.
 		@return: The selection.
 		@rtype: L{textInfos.TextInfo}
 		"""
 		try:
 			return self.obj.makeTextInfo(textInfos.POSITION_SELECTION)
 		except:
+#			try:
+#				return self.obj.makeTextInfo(textInfos.POSITION_CARET)
+#			except:

```

Comment:
Please remove these commented lines. To answer your question, you can discard POSITION_CARET because POSITION_SELECTION should return the caret if there's no selection. Speech relies on this assumption as well.